improve modal styles + add a customisation sample

tobi-or-not-tobi opened the pull request.

Commit:
improve modal styles + add a customisation sample

Matejk00 committed:
refactor: add namespace for ngrx utils stuff 

Closes #5588

Pucek9 committed:
fix: Update migration guide about BillingAddressFormModule #7250 (#7270)

Closes #7250

Pucek9 committed:
fix: Missing translation for "invalid coupon" message #7249 (#7274)

Closes #7249

LTiger14 commented on projects/storefrontstyles/scss/components/product/list/_facet-list.scss:
```diff
@@ -1,3 +1,52 @@
+%cx-dialog {
```
This is a good direction. We should have a global classes or placeholders to use like you did.

LTiger14 commented on projects/storefrontstyles/scss/components/product/list/_facet-list.scss:
```diff
@@ -1,3 +1,52 @@
+}
+
+%cx-overlay {
+  @extend %cx-modal;
```
I guess you could have the overlay as the dialog background like you did or you could have a different element that with this style.

LTiger14 commented on projects/storefrontstyles/scss/components/product/list/_facet-list.scss:
```diff
@@ -12,24 +61,21 @@
-      transform: translate3d(0, 0, 0);
+  &.dialog {
+    @extend %cx-modal;
+    > .inner {
```
I guess the width should be dependent on the screen size

LTiger14 commented on projects/storefrontapp/src/styles.scss:
```diff
@@ -1,3 +1,17 @@
 @import 'storefrontstyles/index';
+
+cx-product-facet-navigation {
+  button.dialog-trigger {
```
Why did you need put this here?

tobi-or-not-tobi replied:
Just to demo that we can influence the experience to force the dialog by showing the button.

LTiger14 commented on projects/storefrontstyles/scss/components/product/list/_facet-list.scss:
```diff
@@ -1,3 +1,52 @@
+%cx-dialog {
```
One small thing, why is the close button on the dialog only visible in mobile/tablet?

tobi-or-not-tobi replied:
in inline mode, you can't close the facets. It's always there ;)

developpeurweb committed:
fix: Adjust translation and position for dontHaveAccount (#7256)

closes GH-3130

developpeurweb committed:
fix: ASM darken background-color of hovered menu items (#7267)

closes GH-5286

developpeurweb committed:
fix: remove redundant styles for order-details page (#7240)

closes GH-5839

developpeurweb committed:
fix: cross browser select menu background color (#7196)

closes GH-3692

developpeurweb committed:
chore: removing duplicated theme selectors (#7195)

closes GH-4614

tobi-or-not-tobi committed:
feat: MediaService takes media threshold for a given media format (#7230)

The media service no longer depends on the breakpoints, and is now using configurable media size information. It will now resolve a media always, either by configuration, best match, or the first format, to avoid an empty image for banners or products.

It is no longer needed to pass in a specific format to the MediaComponent, but it won't break if you still do so. Spartacus will not pass in a format anymore.

closes #7229, fixes #6629, fixes #5398, fixes #5395

tobi-or-not-tobi committed:
feat: new Facet navigation (#7089)

The facet navigation went through a complete overhaul. The existing facet navigation had a number of short comings and some bugs as well. Therefore we decided to rewrite the implementation and address most of the shortcomings for 2.0.

closes #6581

znikola committed:
feat: add threshold option support to intersection service  (#7288)

This PR adds the threshold option support to the intersection service.

dunqan committed:
feat: Refactor `PageMetaResolver` to make it consistent with Handlers implementation (#7282)

PageMetaResolver now implements Handler and shares common interface and logic for handler resolution.

Closes #7279

znikola committed:
fix: schematics - update css docs link (#7286)

This PR updates the CSS docs link.

dunqan committed:
feat: Refactor `LaunchRenderStrategy` to make it consistent with Handlers implementation (#7284)

LaunchRenderStrategy now implements Handler and shares common interface and logic for handler resolution.

Closes #7281

dunqan committed:
feat:: Implement configurable page reload strategy (#7272)

Provide `routing.loadStrategy` configuration. 

Closes #7111

dunqan committed:
feat: Refactor `HttpErrorHandler` to make it consistent with Handlers implementation (#7283)

HttpErrorHandlernow implements Handler and shares common interface and logic for handler resolution.

Closes #7280

Commit:
toggle modal class on body

Commit:
support inline scroll in the facet list

Commit:
Merge branch 'develop' into feature/GH-6581-modal

tobi-or-not-tobi commented:
closing as we already merged the target PR
